apps/gdal_rasterize_lib.cpp

```diff
@@ -50,17 +50,338 @@
 #include "ogr_api.h"
 #include "ogr_core.h"
 #include "ogr_srs_api.h"
+#include "gdalargumentparser.h"
 
 /************************************************************************/
-/*                       ArgIsNumericRasterize()                        */
+/*                          GDALRasterizeOptions()                      */
 /************************************************************************/
 
-static bool ArgIsNumericRasterize(const char *pszArg)
+struct GDALRasterizeOptions
+{
+    std::vector<int> anBandList;
+    std::vector<double> adfBurnValues;
+    bool bInverse = false;
+    std::string osFormat;
+    bool b3D = false;
+    GDALProgressFunc pfnProgress = GDALDummyProgress;
+    void *pProgressData;
+    std::vector<std::string> aosLayers;
+    std::string osSQL;
+    std::string osDialect;
+    std::string osBurnAttribute;
+    std::string osWHERE;
+    CPLStringList aosRasterizeOptions;
+    CPLStringList aosTO;
+    double dfXRes;
+    double dfYRes;
+    CPLStringList aosCreationOptions;
+    GDALDataType eOutputType;
+    std::vector<double> adfInitVals;
+    std::string osNoData;
+    OGREnvelope sEnvelop;
+    int nXSize, nYSize;
+    OGRSpatialReference oOutputSRS;
+
+    bool bTargetAlignedPixels = false;
+    bool bCreateOutput = false;
+};
+
+/************************************************************************/
+/*                     GDALRasterizeOptionsGetParser()                  */
+/************************************************************************/
+
+static std::unique_ptr<GDALArgumentParser>
+GDALRasterizeOptionsGetParser(GDALRasterizeOptions *psOptions,
+                              GDALRasterizeOptionsForBinary *psOptionsForBinary)
+{
+    auto argParser = std::make_unique<GDALArgumentParser>(
+        "gdal_rasterize", /* bForBinary=*/psOptionsForBinary != nullptr);
+
+    argParser->add_description(_("Burns vector geometries into a raster."));
+
+    argParser->add_epilog(
+        _("This program burns vector geometries (points, lines, and polygons) "
+          "into the raster band(s) of a raster image."));
+
+    argParser->add_argument("-b")
+        .metavar("<band>")
+        .append()
+        .scan<'i', int>()
+        .nargs(argparse::nargs_pattern::at_least_one)
+        .store_into(psOptions->anBandList)
+        .help(_("The band(s) to burn values into."));
+
+    argParser->add_argument("-i")
+        .flag()
+        .store_into(psOptions->bInverse)
+        .help(_("Invert rasterization."));
+
+    argParser->add_argument("-at")
+        .flag()
+        .action(
+            [psOptions](const std::string &)
+            {
+                psOptions->aosRasterizeOptions = CSLSetNameValue(
+                    psOptions->aosRasterizeOptions, "ALL_TOUCHED", "TRUE");
+            })
+        .help(_("Enables the ALL_TOUCHED rasterization option."));
+
+    // Mutex for burn/3d/a
```

Comment:
"mutex" has a strong connotation towards threading, perhaps:
```suggestion
    // Mutually exclusive options: -burn, -3d, -a
```
Same remark for below